srml/council/src/lib.rs

```diff
@@ -14,7 +14,159 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-//! Council system: Handles the voting in and maintenance of council members.
+//! # Council Module
+//!
+//! The Council module provides tools to manage the council and proposals.
+//!
+//! - **Seats**
+//! 	- [`seats::Trait`](./seats/trait.Trait.html)
+//! 	- [`Call`](./seats/enum.Call.html)
+//! 	- [`Module`](./seats/struct.Module.html)
+//! - **Motions**
+//! 	- [`motions::Trait`](./motions/trait.Trait.html)
+//! 	- [`Call`](./motions/enum.Call.html)
+//! 	- [`Module`](./motions/struct.Module.html)
+//! - **Voting**
+//! 	- [`voting::Trait`](./voting/trait.Trait.html)
+//! 	- [`Call`](./voting/enum.Call.html)
+//! 	- [`Module`](./voting/struct.Module.html)
+//!
+//! ## Overview
+//!
+//! The Council module provides functionality to handle:
+//!
+//! - The voting in and maintenance of council members.
+//! - Proposing, vetoing, and passing motions.
+//!
+//! The Council's role is to represent passive stakeholders. The Council is an on-chain entity comprised of
+//! a set of account IDs. Its primary tasks are to propose sensible referenda and thwart any uncontroversially
+//! dangerous or malicious referenda.
+//!
+//! ### Terminology
+//!
+//! #### Council Proposals
+//!
+//! - **Council motion:** A mechanism used to enact a proposal.
+//! - **Council origin:** The council (not root) that contains the council motion mechanism.
+//! - **Proposal validity:** A council proposal is valid when it's unique, hasn't yet been vetoed, and
+//! when the council term doesn't expire before the block number when the proposal's voting period ends.
+//! - **Proposal postponement:** Councillors that abstain from voting may postpone a council proposal from
+//! being approved or rejected. Postponement is equivalent to a veto, which only lasts for the cooloff period.
+//! - **Cooloff period:** Period, in blocks, for which a veto is in effect.
+//!
+//! #### Council Proposal Voting
+//!
+//! - **Proposal:** A submission by a councillor. An initial vote of yay from that councillor is applied.
+//! Unlike the Democracy and Treasury modules, the `Proposal` type is very generic.
```

Comment:
```suggestion
//! Unlike the Democracy and Treasury modules, the `Proposal` type is generic and can be anything that can be dispatched (see `type Proposal` in [`motion.rs`]).
```

 (edit suggestion is not ideal but you get the point.)